fix(plugin/vite-resolve): try original extension before replacing with TS variants

sapphi-red opened the pull request:
This is the reason why Vitest is failing on ecosystem-ci.
https://github.com/vitejs/vite-ecosystem-ci/actions/runs/18183865699/job/51764586763#step:7:1878

sapphi-red commented:
* **#6421** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6421?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6421?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6421).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 commented on crates/rolldown_plugin_vite_resolve/src/resolver.rs:
```diff
@@ -156,11 +156,12 @@ fn get_resolve_options(
-      (".jsx".to_string(), vec![".ts".to_string(), ".tsx".to_string(), ".jsx".to_string()]),
-      (".mjs".to_string(), vec![".mts".to_string(), ".mjs".to_string()]),
-      (".cjs".to_string(), vec![".cts".to_string(), ".cjs".to_string()]),
+      (".js".to_string(), vec![".js".to_string(), ".ts".to_string(), ".tsx".to_string()]),
```
LOL. I fell into this before.

shulaoda commented on packages/rolldown/src/binding.js:
```diff
@@ -617,3 +617,4 @@ export { initTraceSubscriber }
 export { registerPlugins }
 export { shutdownAsyncRuntime }
 export { startAsyncRuntime }
+export { JsWatcher }
```
`JsWatcher` has been removed — you need to run `cargo clean` and then execute `just build` again.

sapphi-red committed:
fix(plugin/vite-resolve): try original extension before replacing with TS variants (#6421)

This is the reason why Vitest is failing on ecosystem-ci.
https://github.com/vitejs/vite-ecosystem-ci/actions/runs/18183865699/job/51764586763#step:7:1878
